docs/how_it_works.md

```diff
@@ -150,6 +150,89 @@ Each implementation defines its own objects in subfolders such as `narwhals._pan
 `narwhals._arrow`, `narwhals._polars`, whereas the top-level modules such as `narwhals.dataframe`
 and `narwhals.series` coordinate how to dispatch the Narwhals API to each backend.
 
+## Mapping from API to implementations
+
+If an end user executes some Narwhals code, such as
+
+```python
+df.select(nw.col("a") + 1)
+```
+then how does that get mapped to the underlying dataframe's native API? Let's walk through
+this example to see.
+
+Things generally go through a couple of layers:
+
+- The user calls some top-level Narwhals API.
+- The Narwhals API forwards the call to a Narwhals-compliant dataframe wrapper, such as
+    - `PandasLikeDataFrame` / `ArrowDataFrame` / `PolarsDataFrame` / ...
+    - `PandasLikeSeries` / `ArrowSeries` / `PolarsSeries` / ...
+    - `PandasLikeExpr` / `ArrowExpr` / `ArrowSeries` / ...
```

Comment:
I was wondering about that `ArrowSeries` mentioned next to `PandasLikeExpr` / `ArrowExpr`. We do have PolarsExpr so it caught my attention, but then as far as I remember Polars is backed by Arrow Memory Arrays, is this why we see here `ArrowSeries` instead of `PolarsExpr`?